authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -34,54 +30,36 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Profile;
 import org.springframework.core.annotation.Order;
-import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
 import org.springframework.security.access.expression.method.MethodSecurityExpressionHandler;
-import org.springframework.security.config.Customizer;
 import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
 import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.security.web.authentication.HttpStatusEntryPoint;
 
 @Configuration
 @EnableWebSecurity
 @EnableMethodSecurity
 @Profile("consolidated-auth")
 public class WebSecurityConfig {
   public static final String SESSION_COOKIE = "camunda-session";
-  private static final int ORDER_WEBAPP_API_PROTECTION = 1;
-  private static final int ORDER_UNPROTECTED_API = 2;
-  private static final int ORDER_LOGIN_LOGOUT_HANDLING = 3;
+
   private static final Logger LOG = LoggerFactory.getLogger(WebSecurityConfig.class);
+  private static final int ORDER_PUBLIC_RESOURCES = 0;
+  private static final int ORDER_WEBAPP_API_PROTECTION = 1;
   private static final String LOGIN_URL = "/login";
   private static final String LOGOUT_URL = "/logout";
   private static final Set<String> API_PATHS = Set.of("/v1/**", "/v2/**");
-  private static final Set<String> UNAUTHENTICATED_PATHS =
```

Comment:
💭 I think we have to decide to either: protect everything by default or actually not taking care of any third-party path like `actuator` here. The current draft follows the latter.

The first is definitely more in the "security-first" mindeset, not allowing accidental exposure of new own or third party paths. On the other hand that will require accumulating all potential paths here, but I imagine it's ok to maintain and mitigates potential unintended unportected exposure.